src/lib/storage/constraints/constraint_utils.cpp

```diff
@@ -150,26 +154,13 @@ bool column_is_unique(const std::shared_ptr<Table>& table, const ColumnID column
   return false;
 }
 
-std::vector<bool> columns_are_unique(const std::shared_ptr<Table>& table) {
+std::vector<bool> unique_columns(const std::shared_ptr<Table>& table) {
```

Comment:
Better, still not super happy. :)

@dey4ss, any ideas?

Reply:
Looking what we are doing here, I probably prefer to keep it very simple.
Instead of a bit vector, we use a deduplicated, sorted `std::vector<ColumnID>` that allows `std::ranges::binary_search()` (basically mimicking a set).

Then, `unique_columns()` is fine and we will never be able to measure the performance difference.

Or, I would go with `unique_columns_bitvector` (or `unique_colums_flags`). Don't like it, but we also shouldn't continue discussing this several days. :D

Reply:
To have a deduplicated, sorted `std::vector<ColumnID>` for the binary search, we would have to create that, which is a bit more code compared to just calling  `column_is_unique` a few times. We would have to duplicate the code from `column_is_unique` to fill it, and then call, sort, unique and erase. Don't get me wrong, its not much code, but it still feels weird to make the logic more complex because of some naming issue.

Reply:
Just a note on fill, sort, unique, erase. For such small things, filling a `std::set` and creating the `std::vector` via `set.begin(), set.end()` should be sufficient